main/src/library/Graph.flix

```diff
@@ -101,4 +101,68 @@ namespace Graph {
         Reachable(n1, n2) :- Reachable(n1, m), Edge(m, n2).
     }
 
+    ///
+    /// Optionally returns the shortest distance from `src` to `dst` in the weighted directed graph `g`.
+    ///
+    pub def distance(src: {src :: t}, dst: {dst :: t}, g: m[(t, d, t)]): Option[d] with Foldable[m], Boxable[t], Boxable[d], Add[d] =
+        let edges = inject g into Edge;
+        let x = query edges, dists(src.src) select x from Distance(n, x) where n == dst.dst;
+        List.minimum(x)
+
+    ///
+    /// Returns the shortest distance from `src` to every other vertex in the weighted directed graph `g`.
+    ///
+    pub def distances(src: t, g: m[(t, d, t)]): Map[t, Option[d]] with Foldable[m], Boxable[t], Boxable[d], Add[d] =
+        let edges = inject g into Edge;
+        let allDists = #{
+            // Collects all the distances `d` from `src` to `n` into a Set.
+            AllDists(n; Set#{d}) :- Distance(n, d).
+        };
+        let reachableDists = query edges, dists(src), allDists select (x, Set.minimum(s)) from AllDists(x; s) |> List.toMap;
+        let unreachableDists = unreachableWithDists(src, g)
+                                |> List.foldRight((x, acc) -> Map.insert(x, None, acc), Map.empty());
+        Map.union(reachableDists, unreachableDists)
+
+    ///
+    /// Helper function for `distance` and `distances`.
+    /// Returns a Datalog program which computes distances to all reachable nodes when given a set of `Edge` facts and a `src`.
+    ///
+    def dists(src: t): #{ Edge(t, d, t) | r } with Boxable[t], Boxable[d], Add[d] = #{
```

Comment:
I get the error:
```
#
# An unexpected error has been detected by the Flix compiler:
#
#   No default or matching definition found for 'Add.add'.
```
if I make this into a private helper function instead of declaring it directly in the `distance` and `distances` functions.

Reply:
No, I still get the same error.